Modules/_collectionsmodule.c

```diff
@@ -1814,12 +1977,14 @@ deque_reviter(dequeobject *deque)
     it = PyObject_GC_New(dequeiterobject, &dequereviter_type);
     if (it == NULL)
         return NULL;
+    deque_lock(deque);
     it->b = deque->rightblock;
     it->index = deque->rightindex;
     Py_INCREF(deque);
```

Comment:
Just to make sure I understand correctly: this Py_INCREF is fine where it is because the call will be holding a reference (so there's no risk of the refcount going to 0), right?